zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -380,15 +448,49 @@ impl FromDisk for AddressBalanceLocation {
         let (balance_bytes, location_bytes) = disk_bytes.as_ref().split_at(BALANCE_DISK_BYTES);
 
         let balance = Amount::from_bytes(balance_bytes.try_into().unwrap()).unwrap();
-        let location = AddressLocation::from_bytes(location_bytes);
+        let address_location = AddressLocation::from_bytes(location_bytes);
 
-        let mut balance_location = AddressBalanceLocation::new(location);
+        let mut balance_location = AddressBalanceLocation::new(address_location);
         *balance_location.balance_mut() = balance;
 
         balance_location
     }
 }
 
+impl IntoDisk for UnspentOutputAddressLocation {
+    type Bytes = Vec<u8>;
+
+    fn as_bytes(&self) -> Self::Bytes {
+        let output_bytes = self.output().zcash_serialize_to_vec().unwrap();
+
+        // If there is no address location, don't write any bytes for it
+        let address_location_bytes = self
+            .address_location()
+            .map(|address_location| address_location.as_bytes().to_vec())
+            .unwrap_or_default();
+
+        [output_bytes, address_location_bytes].concat()
```

Comment:
Idea: this should eliminate an extra `Vec` allocation, but I'm not sure if:

a) it negatively impacts readability
b) the compiler optimizes this anyway

```suggestion
        let mut bytes = self.output().zcash_serialize_to_vec().unwrap();

        // If there is no address location, don't write any bytes for it
        if let Some(address_location) = self.address_location() {
            bytes.extend(address_location.as_bytes())
        }

        bytes
```

Reply:
Thanks for the suggestion, I think it is clearer.